src/main/java/com/ericsson/ei/notifications/HttpRequest.java

```diff
@@ -154,45 +157,36 @@ private void addAuthenticationData() throws AuthenticationException {
         String username = subscriptionField.get("userName");
         String password = subscriptionField.get("password");
 
-        boolean authenticationDetailsProvided = isAuthenticationDetailsProvided(authType, username,
-                password);
+        boolean authenticationDetailsProvided = EncryptionFormatter.verifyAuthenticationDetails(
+                authType, username, password);
         if (!authenticationDetailsProvided) {
             return;
         }
+        if (encryptor != null && encryptor.isJasyptPasswordSet()
+                && EncryptionFormatter.isEncrypted(password)) {
+            password = doDecryption(password);
+        }
 
         String encoding = Base64.getEncoder()
                                 .encodeToString((username + ":" + password).getBytes());
         this.headers.add("Authorization", "Basic " + encoding);
         LOGGER.debug("Successfully added header for 'Authorization'");
 
-        if (authType.equals(AUTHENTICATION_TYPE_BASIC_AUTH_JENKINS_CSRF)) {
+        if (authType.equals(AuthenticationType.BASIC_AUTH_JENKINS_CSRF.getValue())) {
             JsonNode crumb = jenkinsCrumb.fetchJenkinsCrumb(encoding, this.url);
             addJenkinsCrumbData(crumb);
         }
 
     }
 
-    /**
-     * Returns a boolean indicating that authentication details was provided in
-     * the subscription.
-     *
-     * @param authType
-     * @param username
-     * @param password
-     * @return
-     */
-    private boolean isAuthenticationDetailsProvided(String authType, String username,
-            String password) {
-        if (authType.isEmpty() || authType.equals(AUTHENTICATION_TYPE_NO_AUTH)) {
-            return false;
-        }
-
-        if (username.isEmpty() && password.isEmpty()) {
-            LOGGER.error("userName/password field in subscription is missing.");
-            return false;
+    private String doDecryption(String password) {
+        try {
+            String encryptedPassword = EncryptionFormatter.removeEncryptionParentheses(password);
+            password = encryptor.decrypt(encryptedPassword);
+        } catch (EncryptorException e) {
+            LOGGER.error("", e);
         }
-
-        return true;
+        return password;
```

Comment:
Maybe make this in a finally statement? :)

Reply:
I don't think you can return in a finally block.